### ISSUE TITLE
Issues with version 25.0.5snap2 after upgrading ubuntu packages

### ISSUE THREAD

**Issue by 7djs7** [nextcloud-snap/nextcloud-snap#2424]
#### Describe the bug
Nextcloud going in and out of maintenance mode 

#### To Reproduce
updated ubuntu 22.04 and snap repositories
snapd:amd64 (2.58+22.04, 2.58+22.04.1)

#### Expected behavior
No maintenance mode issues .

#### OS/snapd/snap version
What OS and version are you running? Please also include the output of the
following commands:

    $ snap list nextcloud

nextcloud  25.0.5snap2  34372  latest/stable  nextcloud✓  

and:

    $ snap version

snap    2.58.3
snapd   2.58.3
series  16
ubuntu  22.04
kernel  5.15.0-73-generic

#### Logs
Please paste in the output from
[this script](https://github.com/nextcloud/nextcloud-snap/wiki/Issue-Debugging-Scripts)
to assist in debugging (note that it's pre-formatted).

**Comment by scubamuc** [comment]
could you post your logs
see here: https://github.com/nextcloud-snap/nextcloud-snap/wiki/Issue-Debugging-Scripts

**Comment by scubamuc** [comment]
> I think it's a mysql issue. 

nope, it seems you're one version behind. so you haven't been bitten by the mysql [bug](#2405). 

your issue seems to be related to the "old faithful" #2288 

your logs:

> `"exception":{"Exception":"GuzzleHttp\\Exception\\ConnectException","Message":"cURL error 6: Could not resolve host: nextcloud.divertas.net (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://nextcloud.divertas.net/extra-apps/richdocumentscode/proxy.php?req=/hosting/capabilities","Code":0,"Trace":[{"file":"/var/snap/nextcloud/34372/nextcloud/extra-apps/signlive/vendor/guzzlehttp/guzzle/src/Handler"`

this is a duplicate of #2288

- remove `richdocumentscode`

Nextcloud-office is incompatipble.

**Comment by scubamuc** [comment]
okay, your logs are showing recurring exceptions regarding 'tensorflow' recognize amongst others.
try removing the recognize app too. you have plenty 3rd party apps installed. 
this doesn't seem to be a snap issue.

**Comment by scubamuc** [comment]
great

please close

**Comment by scubamuc** [comment]
https://apps.nextcloud.com/apps/recognize

> System with glibc (usually the norm on Linux; FreeBSD, Alpine linux and thus also the official Nextcloud Docker container and Nextcloud AIO are not such systems)

this applies to Nextcloud-snap too

recognize app is not compatible

**Comment by szaimen** [comment]
You can still run recognize on AIO only with subnative speed:
![image](https://github.com/nextcloud-snap/nextcloud-snap/assets/42591237/e6f3e42c-0333-4dc1-94a1-008c494c8064)

**Comment by scubamuc** [comment]
@kyrofa 
i didn't see these exceptions in the first mention of the mysqld [bug](https://github.com/nextcloud-snap/nextcloud-snap/issues/2424#2405).

but some folks are mentioning that recognize app was the cause of their issues... [here](https://github.com/nextcloud-snap/nextcloud-snap/issues/2405#issuecomment-1579048218) and [here](https://github.com/nextcloud-snap/nextcloud-snap/issues/2405#issuecomment-1579056729) could it be possible that your headache is connected to a 3rd party bug and not because of a mysqld bug?

https://github.com/nextcloud/recognize/issues/113

**Comment by kyrofa** [comment]
Both of those comments refer to 25.0.5snap2 though, which doesn't have the new mysql in it. I suspect these are different issues.

**Comment by scubamuc** [comment]
> Both of those comments refer to 25.0.5snap2 though, which doesn't have the new mysql in it.

yeah, kind of... both reverted to 25.0.5snap2, removed the app. one upgraded successfully, the other hasn't tried.

**Comment by vawaver** [comment]
@kyrofa 
I just did the same process on my second Nextcloud snap installation on another server.

1. Removed Recognize app
2. Update 25.0.5snap2  -> 25.0.6snap1 
3. Success - all working fine.

**Comment by scubamuc** [comment]
@vawaver thanks for confirming.  if @7djs7 would give it a shot that would be a great confirmation

@kyrofa is suspicious and he sure has the deepest insight. i'd only rely on his final verdict,

> Both of those comments refer to 25.0.5snap2 though, which doesn't have the new mysql in it. I suspect these are different issues.

although both reverted to 25.0.5snap2 after unsuccessful upgrade, both removed the `recognize` app. two successful upgrades have been confirmed until now

my instances never had `recognize` installed so maybe that's why I've never faced the issue. we'll need some further confirmation from other folks.

so we'll need to wait and see...

**Comment by 7djs7** [comment]
@vawaver I can upgrade back to 25.0.6snap1. Is it with command "sudo snap refresh nextcloud --stable" ?

**Comment by 7djs7** [comment]
@scubamuc @kyrofa  Done. Upgraded to 25.0.6snap1. So far no issues for the past 20 minutes. It's been stable, not going in and out of maintenance mode.

**Comment by scubamuc** [comment]
@7djs7 thanks for confirming. that's the third successful upgrade to 25.06snap1 after reverting to 25.05snap2 due to #2405 and removing `recognize`

**Comment by YoannPa** [comment]
To add to previous testimonies, I also experienced the in and out maintenance mode bug in `26.0.3snap1`
After disabling the app **recognize** (and actually removing it definitely right after deactivation) from the interface, I did:
```
sudo nextcloud.occ maintenance:mode --on
sudo nextcloud.occ maintenance:repair
sudo nextcloud.occ maintenance:mode --off
sudo snap restart nextcloud
```
Not sure all the commands are really necessary (I guess the maintenance:repair is the most important).
So far, the problem is solved, and I did not had to revert to the previous version.

So, maybe reverting to the previous version is not always necessary.
Maybe what is really necessary is to deactivate the recognize app, and to `nextcloud.occ maintenance:repair` ?

Thank you very much anyway for the issue, it was extremely helpful to find out what the problem was.

**Comment by 7djs7** [comment]
> To add to previous testimonies, I also experienced the in and out maintenance mode bug in `26.0.3snap1` After disabling the app **recognize** (and actually removing it definitely right after deactivation) from the interface, I did:
> 
> ```
> sudo nextcloud.occ maintenance:mode --on
> sudo nextcloud.occ maintenance:repair
> sudo nextcloud.occ maintenance:mode --off
> sudo snap restart nextcloud
> ```
> 
> Not sure all the commands are really necessary (I guess the maintenance:repair is the most important). So far, the problem is solved, and I did not had to revert to the previous version.
> 
> So, maybe reverting to the previous version is not always necessary. Maybe what is really necessary is to deactivate the recognize app, and to `nextcloud.occ maintenance:repair` ?
> 
> Thank you very much anyway for the issue, it was extremely helpful to find out what the problem was.

Once I removed the recognize app as well, my issues were solved and was able to upgrade back to 25.0.6snap1 with no issues and have had no issues since.

**Comment by scubamuc** [comment]
Adding this for reference: https://github.com/nextcloud/recognize/issues/890
